src/cell.cpp

```diff
@@ -1412,7 +1465,14 @@ std::unordered_map<int32_t, vector<int32_t>> Cell::get_contained_cells(
     return contained_cells;
 
   // find the pathway through the geometry to this cell
-  vector<ParentCell> parent_cells = this->find_parent_cells(instance);
+  vector<ParentCell> parent_cells;
+
+  // if a positional hint is provided, attempt to do a fast lookup
+  // of the parent cells
+  if (hint)
+    parent_cells = find_parent_cells(instance, *hint);
+  else
+    parent_cells = exhaustive_find_parent_cells(instance);
```

Comment:
Could combine this whole thing: `vector<ParentCell> parent_cells = hint ? ... : ...;`